File: algorithms/2_medium/U/00062_unique_paths/math.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        mx = m - 1 if m > n else n - 1
        mn = m - 1 if m < n else n - 1
        high = mx + mn
        res = 1

        for k in range(mx + 1, high + 1):
            res *= k

        for k in range(2, mn + 1):
            res //= k

        return res

File: algorithms/1_easy/A/00258_add_digits/math.rs
impl Solution {
    pub fn add_digits(num: i32) -> i32 {
        debug_assert!(num >= 0);
        if num == 0 {
            return 0;
        }

        let res = num % 9;
        if res == 0 { 9 } else { res }
    }
}
